lld/ELF/InputSection.cpp

```diff
@@ -947,12 +953,14 @@ void InputSection::relocateNonAlloc(uint8_t *buf, ArrayRef<RelTy> rels) {
       //
       // TODO To reduce disruption, we use 0 instead of -1 as the tombstone
       // value. Enable -1 in a future release.
+
+      // Extend to 64bit MAX for 64 bit relocations, LocalTU, in .debug_names.
+      if (isDebugNames && type == target.symbolicRel)
+        tombstone = SignExtend64<32>(*tombstone);
```

Comment:
I guess this only handles one case of 64 bit relocations, but there might be others? & I'm not sure `isDebugNames` is the special case here - shouldn't it be known by the relocation, rather than the section the relocation is in?

Reply:
ok, made it more generic.